src/extension/alterschema/linter/metaschema_uri_missing_hash.h

```diff
@@ -0,0 +1,45 @@
+class MetaschemaUriMissingHash final : public SchemaTransformRule {
+public:
+  MetaschemaUriMissingHash()
+      : SchemaTransformRule{"metaschema_uri_missing_hash",
+                            "The canonical metaschema URIs end with `schema#`. "
+                            "Omitting the fragment identifier can confuse "
+                            "tooling and may bypass caching"} {};
+
+  [[nodiscard]] auto condition(const sourcemeta::core::JSON &schema,
+                               const sourcemeta::core::JSON &,
+                               const sourcemeta::core::Vocabularies &,
+                               const sourcemeta::core::SchemaFrame &,
+                               const sourcemeta::core::SchemaFrame::Location &,
+                               const sourcemeta::core::SchemaWalker &,
+                               const sourcemeta::core::SchemaResolver &) const
+      -> sourcemeta::core::SchemaTransformRule::Result override {
+    if (!schema.is_object() || !schema.defines("$schema") ||
+        !schema.at("$schema").is_string()) {
+      return false;
+    }
+
+    const auto schema_value = schema.at("$schema").to_string();
```

Comment:
```suggestion
    const auto &schema_value{schema.at("$schema").to_string()};
```

Otherwise you are unnecessarily copying the string